### PR TITLE
fix: simplify APIEndpoints() usage

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: simplify APIEndpoints() usage

## Motivation and Context
improvements include

- skip IPv6 correctly
- do not set a default value for
  MINIO_SERVER_URL, let it be
  configured if not use local IPs

Bonus:

- In healing return error from listPathRaw()
- update console to v0.8.3

## How to test this PR?
Nothing special set environment variables like and see what is printed on the banner
```
MINIO_SERVER_URL
MINIO_BROWSER_REDIRECT_URL
```

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by poornas — APPROVED** [review]

File: cmd/common-main.go [hunk 1]
```diff
@@ -116,10 +116,11 @@ const consolePrefix = "CONSOLE_"
 func minioConfigToConsoleFeatures() {
 	os.Setenv("CONSOLE_PBKDF_SALT", globalDeploymentID)
 	os.Setenv("CONSOLE_PBKDF_PASSPHRASE", globalDeploymentID)
-	if globalMinioEndpoint == "" {
-		logger.Fatal(errInvalidArgument, "Unable to start console service MinIO Endpoint is empty")
+	if globalMinioEndpoint != "" {
+		os.Setenv("CONSOLE_MINIO_SERVER", globalMinioEndpoint)
+	} else {
+		os.Setenv("CONSOLE_MINIO_SERVER", getAPIEndpoints()[0])
```
Comment (krisis) [inline cmd/common-main.go:122]: wouldn't it make sense to set `CONSOLE_MINIO_SERVER` to `globalEndpoints.Localhost()`?

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/server-startup-msg.go [hunk 2]
```diff
@@ -79,39 +80,41 @@ func printStartupMessage(apiEndpoints []string, err error) {
...
 	h, _, err := net.SplitHostPort(host)
 	if err != nil {
 		h = host
 	}
 	ip := net.ParseIP(h)
-	ok := ip.To4() != nil // This is always true of IP is IPv4
-
-	// Returns true if input is not IPv4.
-	return !ok
+	return ip.To16() != nil && ip.To4() == nil
```
Comment (krisis) [inline cmd/server-startup-msg.go:90]: ip.To16() returning non-nil doesn't imply that it is an IPV6 address.
> a 16-byte slice can still be an IPv4 address. 
ref: https://pkg.go.dev/net#IP

File: cmd/server-startup-msg.go [hunk 3]
```diff
@@ -79,39 +80,41 @@ func printStartupMessage(apiEndpoints []string, err error) {
...
 // strip api endpoints list with standard ports such as
 // port "80" and "443" before displaying on the startup
 // banner.  Returns a new list of API endpoints.
 func stripStandardPorts(apiEndpoints []string, host string) (newAPIEndpoints []string) {
-	if len(apiEndpoints) == 1 && globalBrowserRedirectURL != nil {
-		if apiEndpoints[0] == globalBrowserRedirectURL.String() {
-			return []string{globalBrowserRedirectURL.String()}
-		}
+	if len(apiEndpoints) == 1 {
+		return apiEndpoints
```
Comment (krisis) [inline cmd/server-startup-msg.go:98]: in this case we don't remove port 80, 443.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/common-main.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/common-main.go:122]: no @krisis it needs to be set to the first entry in the getAPIEndpoints() - we do not know Localhost() has the certs.

File: cmd/server-startup-msg.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/server-startup-msg.go:90]: that is why To4() needs to be nil @krisis

File: cmd/server-startup-msg.go (hunk 3 above)
Comment (harshavardhana) [reply cmd/server-startup-msg.go:98]: That is not removed on purpose since it's custom settings @krisis - we rely on the user to set it right.

**Review by krisis — APPROVED** [review]

**Pull request merged**